[BUG] macOS debug build broken

**Describe the bug**
PR #816 seems to have broken the debug build, which will not complete when using Clang 21 on macOS. The culprit is the following line:

https://github.com/revbayes/revbayes/blob/9b2242032ea7838988d1922c456666ff64986501/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h#L240

```
/Users/David/revbayes/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h:240:27: error: no viable overloaded '='
  240 |     assert(pl_left.dims() = pl_right.dims());
      |            ~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/assert.h:72:25: note: expanded from macro 'assert'
   72 |     (__builtin_expect(!(e), 0) ? __assert_rtn(__func__, __ASSERT_FILE_NAME, __LINE__, #e) : (void)0)
      |                         ^
```

**To reproduce**
```
git clone https://github.com/revbayes/revbayes.git
cd revbayes/projects/cmake
./build.sh -debug true -DCMAKE_APPLE_SILICON_PROCESSOR="arm64" -j 12
```

**Expected behavior**
It should build.

**Computer info**
- OS: macOS 26.3 (Tahoe)
- Architecture: ARM64
- Build system: CMake 4.0.2
- Compiler: AppleClang 21.0.0.21000099
- Boost: 1.88.0
- RevBayes version: 1.3.2, current tip of development (9b22420).

--- a/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h
+++ b/src/core/distributions/phylogenetics/substitution/PhyloCTMCSiteHomogeneousNucleotide.h
@@ -237,7 +237,7 @@ void RevBayesCore::PhyloCTMCSiteHomogeneousNucleotide<charType>::computeInternal
     auto& pl_right = this->getPartialLikelihoodsForNode(right);
     const double*   p_left  = pl_left.likelihoods.data();
     const double*   p_right = pl_right.likelihoods.data();
-    assert(pl_left.dims() = pl_right.dims());
+    assert(pl_left.dims() == pl_right.dims());
 
     double*         p_node  = this->createEmptyPartialLikelihoodsForNode(node_index, pl_left.dims()).likelihoods.data();
 

